customerio/api.py

```diff
@@ -4,11 +4,15 @@
 import base64
 import json
 from .client_base import ClientBase, CustomerIOException
+from .regions import Regions, Region
 
 
 class APIClient(ClientBase):
-    def __init__(self, key, url=None, retries=3, timeout=10, backoff_factor=0.02):
-        self.url = url or 'https://api.customer.io'
+    def __init__(self, key, url=None, region=Regions.US, retries=3, timeout=10, backoff_factor=0.02):
+        if not isinstance(region, Region):
+            raise CustomerIOException('region must be of type {region}'.format(region=Region))
```

Comment:
i tried this out and the error message does not seem great
```python
>>> from collections import namedtuple
>>> Region = namedtuple('Region', ['name', 'track_host', 'api_host'])
>>> print('region must be of type {region}'.format(region=Region))
region must be of type <class '__main__.Region'>
```

maybe the error should just be `invalid region provided`.